src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/SourceGeneratorAttributeAnalyzerFix.cs

```diff
@@ -0,0 +1,111 @@
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic, Name = nameof(SourceGeneratorAttributeAnalyzerFix))]
+    [Shared]
+    public sealed class SourceGeneratorAttributeAnalyzerFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(
+            DiagnosticIds.MissingSourceGeneratorAttributeId);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span);
+
+            if (node is null)
+            {
+                return;
+            }
+
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                AddFix(
+                    string.Format(CodeAnalysisDiagnosticsResources.AddGeneratorAttribute_1, LanguageNames.CSharp),
+                    context, document, node, diagnostic, LanguageNames.CSharp);
+
+                AddFix(
+                    string.Format(CodeAnalysisDiagnosticsResources.AddGeneratorAttribute_1, LanguageNames.VisualBasic),
+                    context, document, node, diagnostic, LanguageNames.VisualBasic);
+
+                AddFix(
+                    string.Format(CodeAnalysisDiagnosticsResources.AddGeneratorAttribute_2, LanguageNames.CSharp, LanguageNames.VisualBasic),
+                    context, document, node, diagnostic, LanguageNames.CSharp, LanguageNames.VisualBasic);
+            }
+        }
+
+        private static void AddFix(string title, CodeFixContext context, Document document, SyntaxNode node, Diagnostic diagnostic, params string[] languageNames)
+        {
+            var codeAction = CodeAction.Create(
+                title,
+                (cancellationToken) => FixDocumentAsync(document, node, languageNames, cancellationToken),
+                equivalenceKey: nameof(SourceGeneratorAttributeAnalyzerFix));
+
+            context.RegisterCodeFix(codeAction, diagnostic);
+        }
+
+        public override FixAllProvider? GetFixAllProvider() => null;
+
+        private static async Task<Document> FixDocumentAsync(Document document, SyntaxNode node, string[] languageNames, CancellationToken cancellationToken)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            var generator = editor.Generator;
+
+            SyntaxNode? generatorAttribute;
+
+            if (languageNames.Length == 1 && languageNames[0] == LanguageNames.CSharp)
+            {
+                // CSharp is the only language, which is the default paramterless 
+                // constructor for the Generator attribute
+                generatorAttribute = generator.Attribute(WellKnownTypeNames.MicrosoftCodeAnalysisGeneratorAttribute);
```

Comment:
See https://github.com/dotnet/roslyn-analyzers/issues/5797:

```suggestion
                generatorAttribute = generator.Attribute(WellKnownTypeNames.MicrosoftCodeAnalysisGeneratorAttribute).WithAddImportsAnnotation();
```